### PR TITLE
chore: fix ts strict errors primitive tests

### PR DESCRIPTION
**Pull request by reesscot**
#### Description of changes
Fixes all `ref.current` ts strict errors in primitive test files

#### Checklist

- [x] PR description included
- [ ] `yarn test` passes
- [ ] Tests are updated
- [ ] No side effects or [`sideEffects`](https://github.com/aws-amplify/amplify-ui/blob/main/packages/react/CONTRIBUTING.md#code-standards) field updated
- [ ] Relevant documentation is changed or added (and PR referenced)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by 0618 — APPROVED** [review]

File: packages/react/src/primitives/ToggleButtonGroup/__tests__/ToggleButtonGroup.test.tsx [hunk 1]
```diff
@@ -15,7 +15,7 @@ describe('ToggleButtonGroup: ', () => {
     const [value, setValue] = React.useState(['test-button-1']);
     return (
       <ToggleButtonGroup
-        onChange={(value: string[]) => setValue(value)}
+        onChange={(value) => setValue(value as string[])}
```
Comment (calebpollman) [inline packages/react/src/primitives/ToggleButtonGroup/__tests__/ToggleButtonGroup.test.tsx:18]: Non-blocking: Kind of think we should just fix the typing of `ToggleButtonGroup`, the `onChange` handler should know the type of param it passes in to `onChange` based on the `value`

**Review by joebuono — APPROVED** [review]
Thanks

**Review by calebpollman — APPROVED** [review]
Left a non-blocking comment but LGTM otherwise

File: packages/react/src/primitives/ToggleButtonGroup/__tests__/ToggleButtonGroup.test.tsx (hunk 1 above)
Comment (reesscot) [reply packages/react/src/primitives/ToggleButtonGroup/__tests__/ToggleButtonGroup.test.tsx:18]: Yeah, I was wondering about this too. I haven't had much luck with conditional types though

**Pull request merged**